src/component/fieldsForBook/FieldsForBook.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { ChangeEvent, FC, useCallback } from 'react'
+import { Button } from '@material-ui/core';
+
+import { TextFields } from '../input/Input'
+import { SelectComponent } from '../select/SelectComponent';
+import { SelectOption } from '../select/SelectOption';
+import { InputUploadImg } from './components/InputUploadImg';
+import { IBook } from '../../interface/IBook';
+import { ErrorType } from '../../utils/ErrorsHanding';
+import { selectOptionCategory, selectOptionLanguage, selectOptionYear } from './constants'
+import { LoadingCircular } from '../../component/loading/LoadingCircular';
+import { SnackbarAccepted } from '../snackbar/SnackbarAccepted';
+import { GoBackPage } from '../goBackPage/GoBackPage';
+
+import './style.css'
+
+type Props = {
+    namePage: string,
+    form: Record<string, keyof Omit<IBook, 'idBook'>>,
+    setFormBook: (target: ChangeEvent<HTMLInputElement>) => void,
+    loading: boolean,
+    errorsValid: ErrorType[] | null,
+    messageCreated: string,
+    isShowSnackbar: boolean,
+    onCloseSnackbar: () => void,
+    handlerCreateBook: () => Promise<void>,
+    buttonCreateName: string,
+    isEdit?: boolean,
+}
+  
+export const FieldsForBook: FC<Props> = ({ namePage, form, isEdit, setFormBook, loading, errorsValid, messageCreated, isShowSnackbar, onCloseSnackbar, handlerCreateBook, buttonCreateName }) => {  
```

Comment:
Rename `FieldsForBook` to `BookForm` and file as well

Reply:
fixed